### PR TITLE
Add Mawranth's Hair stuff

### PR DESCRIPTION
**Pull request by Mawranth**
#### Summary
Multiple hair styles & colour variations.

#### Content of the change

Changes to JSON data files and the individual PNGs of the hair styles.

#### Additional information
I've never done this before. Let me know if I did anything wrong.

### CODE REVIEW

File: gfx/MShockXotto+/pngs_tiles_32x32/character/overlay/mutations/hair/overlay_male_mutation_hair.json [hunk 1]
```diff
@@ -616,5 +616,318 @@
     "fg": [ "overlay_mutation_long_over_eye_var_white" ],
     "rotates": false,
     "bg": [  ]
-  }
+  },
+  {
+	  "id": "overlay_male_mutation_hair_beaded_var_black",
+	  "fg": [ "overlay_male_mutation_hair_beaded_var_black" ],
```
[the remark this answers is not in the document]
Comment (Mawranth) [reply gfx/MShockXotto+/pngs_tiles_32x32/character/overlay/mutations/hair/overlay_male_mutation_hair.json]: Okay. I removed the files. I do not have a PR to the main game repo,

File: gfx/MShockXotto+/pngs_tiles_32x32/character/overlay/mutations/hair/overlay_female_mutation_hair.json [hunk 2]
```diff
@@ -1,620 +0,0 @@
-[
-  {
-    "id": "overlay_female_mutation_hair_crewcut_var_black",
-    "fg": [ "overlay_female_mutation_hair_black_crewcut" ],
```
Comment (Fris0uman) [inline gfx/MShockXotto+/pngs_tiles_32x32/character/overlay/mutations/hair/overlay_female_mutation_hair.json:4]: this part should not have been removed, as you can see the id and fg field there are not the same.

**Comment by Fris0uman** [comment]
I'll merge the sprites into the tileset if/when the hair styles are actually added to the game

**Comment by Mawranth** [comment]
> I'll merge the sprites into the tileset if/when the hair styles are actually added to the game

That's fine. It's not like this is going anywhere.

**Comment by BalthazarArgall** [comment]
> I'll merge the sprites into the tileset if/when the hair styles are actually added to the game

Not merged yet but the PR adding them to the game can be found [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/73465) if you want to check that everything is set up correctly.

**Review by LyleSY — CHANGES_REQUESTED** [review]
The 1,240 lines of removed JSON need to be restored before this is merged. Otherwise I believe this is ready since the matching PR has been merged in DDA

**Comment by Fris0uman** [comment]
Merged in #2372

**Pull request closed without merging**